DataFormats/Common/interface/DetSetVectorNew.h

```diff
@@ -23,6 +23,8 @@ namespace edm { namespace refhelper { template<typename T> struct FindForNewDetS
 namespace edmNew {
   typedef uint32_t det_id_type;
 
+  struct CapacityExaustedException {};
```

Comment:
If you expect the exception to happen so often that making a string is a performance problem then shouldn't be avoiding using exceptions?

Also, I understand you do not intend for the exception to ever spill out but since the exception happens in public interfaces there is no guarantee it won't spill out.